src/test/groovy/pl/pandziak/ticketbookingapp/api/ReservationEndpointSpec.groovy

```diff
@@ -0,0 +1,197 @@
+package pl.pandziak.ticketbookingapp.api
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import pl.pandziak.ticketbookingapp.ErrorResponse
+import pl.pandziak.ticketbookingapp.TicketBookingApplicationSpec
+import pl.pandziak.ticketbookingapp.domain.Room
+import pl.pandziak.ticketbookingapp.domain.TicketPricing
+import pl.pandziak.ticketbookingapp.domain.reservation.ReservationFacade
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.ReservationRequestDto
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.ReservationResponseDto
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.TicketDto
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.TicketsDto
+import pl.pandziak.ticketbookingapp.domain.screening.Screening
+import pl.pandziak.ticketbookingapp.domain.screening.ScreeningFacade
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.ScreeningResponseDto
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.SeatDto
+
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+import java.util.stream.Collectors
+
+class ReservationEndpointSpec extends TicketBookingApplicationSpec {
+
+    @Autowired
+    private ReservationFacade reservationFacade
+
+    @Autowired
+    private ScreeningFacade screeningFacade
+
+    def setup() {
+        reservationRepository.deleteAll()
+    }
+
+    def "should create reservation"() {
+        given:
+            Screening screening = createScreening()
+            ReservationRequestDto reservationRequest = prepareReservationRequest(screening.getId())
+
+        and:
+            URI uri = getBaseUri()
+                        .path("reservations")
+                        .build()
+                        .toUri()
+
+        when:
+            ResponseEntity<ReservationResponseDto> result = httpClient.postForEntity(
+                    uri,
+                    reservationRequest,
+                    ReservationResponseDto.class)
+
+        then:
+            result.statusCode == HttpStatus.CREATED
+            with (result.body) {
```

Comment:
generowałeś equalsy do dto i w sumie to chyba jedyne miejsce gdzie powinieneś ich użyć? :) bo inaczej po co Ci one? :)

Reply:
Prawda. W czasie developmentu aplikacji porównywałam całe obiekty i wtedy te metody miały sens. Po refaktorze na 'witha' i porównywanie pole po polu już nie pamiętałam żeby usunąć te nadmiarowo nadpisane metody.